pkg/app/server/service/webservice/service.pb.auth.go

```diff
@@ -83,156 +79,141 @@ func (a *authorizer) getRBAC(ctx context.Context, projectID string) (*rbac, erro
 		return nil, err
 	}
 
-	v := &rbac{p.RbacRoles}
-	if err = a.rbacCache.Put(projectID, v); err != nil {
+	if err = a.rbacCache.Put(projectID, p.RbacRoles); err != nil {
 		a.logger.Warn("unable to store the rbac in memory cache",
 			zap.String("project", projectID),
 			zap.Error(err),
 		)
 	}
-	return v, nil
-}
-
-type rbac struct {
-	Roles []*model.ProjectRBACRole
-}
 
-func (r *rbac) HasPermission(typ model.ProjectRBACResource_ResourceType, action model.ProjectRBACPolicy_Action) bool {
-	for _, v := range r.Roles {
-		if v.HasPermission(typ, action) {
-			log.Printf("this rbac has a permission: %v", r.Roles)
-			return true
+	rs := make(map[string]*model.ProjectRBACRole, len(p.RbacRoles))
+	for _, v := range p.RbacRoles {
+		rs[v.Name] = v
+	}
+	ret := make([]*model.ProjectRBACRole, 0, len(roles))
+	for _, r := range roles {
+		if v, ok := rs[r]; ok {
+			ret = append(ret, v)
 		}
 	}
-	log.Printf("this rbac does not have a permission: %v", r.Roles)
-	return false
+	return ret, nil
 }
 
-func (r *rbac) FilterByNames(names []string) *rbac {
-	roles := make([]*model.ProjectRBACRole, 0, len(names))
-	rs := make(map[string]*model.ProjectRBACRole, len(r.Roles))
-	for _, v := range r.Roles {
-		rs[v.Name] = v
-	}
-	for _, n := range names {
-		if v, ok := rs[n]; ok {
-			roles = append(roles, v)
+func verifyRBACPermission(typ model.ProjectRBACResource_ResourceType, action model.ProjectRBACPolicy_Action, roles []*model.ProjectRBACRole) bool {
+	for _, r := range roles {
+		if r.HasPermission(typ, action) {
+			return true
 		}
 	}
-	r.Roles = roles
-	return r
+	return false
 }
 
 // Authorize checks whether a role is enough for given gRPC method or not.
 func (a *authorizer) Authorize(ctx context.Context, method string, r model.Role) bool {
-	rbac, err := a.getRBAC(ctx, r.ProjectId)
+	roles, err := a.getRBACRoles(ctx, r.ProjectId, r.ProjectRbacRoles)
```

Comment:
this `getRBACRoles` is a bit odd to me, the return is not RBACRoles but actually `r.ProjectRbacRoles` or ProjectRBACRoles, so how about renaming it to something like `getProjectRbacRoles`?

Reply:
@nghialv @khanhtc1202 
how about like this?
```
func (a *authorizer) getProjectRBACRoles(ctx context.Context, projectID string, roles []string) ([]*model.ProjectRBACRole, error) {
	all, err := a.getAllProjectRBACRoles(ctx, projectID)
	if err != nil {
		return nil, err
	}
	rs := make(map[string]*model.ProjectRBACRole, len(all))
	for _, v := range all {
		rs[v.Name] = v
	}
	ret := make([]*model.ProjectRBACRole, 0, len(roles))
	for _, r := range roles {
		if v, ok := rs[r]; ok {
			ret = append(ret, v)
		}
	}
	return ret, nil
}

func (a *authorizer) getAllProjectRBACRoles(ctx context.Context, projectID string) ([]*model.ProjectRBACRole, error) {
	if _, ok := a.projectsInConfig[projectID]; ok {
		p := &model.Project{Id: projectID}
		p.SetBuiltinRBACRoles()
		return p.RbacRoles, nil
	}

	if v, err := a.rbacCache.Get(projectID); err == nil {
		return v.([]*model.ProjectRBACRole), nil
	}

	p, err := a.projectStore.Get(ctx, projectID)
	if err != nil {
		a.logger.Error("failed to get project",
			zap.String("project", projectID),
			zap.Error(err),
		)
		return nil, err
	}

	if err = a.rbacCache.Put(projectID, p.RbacRoles); err != nil {
		a.logger.Warn("unable to store the rbac in memory cache",
			zap.String("project", projectID),
			zap.Error(err),
		)
	}
	return p.RbacRoles, nil
}
```

Reply:
LGTM 👍